### PR TITLE
[script.artistslideshow] v2.1.0

### PR DESCRIPTION
**Pull request by pkscout**
### Description
- updated AS to save and use unhashed artist and image names
- created upgrade routine to rename hashed artist and image names
- included new option to ONLY use local artist path for storage
- cleaned up folder path creation routines
- removed old download exclusion logic in favor of new _imgdb.nfo history
### Checklist:

- [x ] My code follows the [add-on rules](http://kodi.wiki/view/Add-on_rules) and [piracy stance](http://kodi.wiki/view/Official:Forum_rules#Piracy_Policy) of this project. 
- [x ] I have read the [CONTRIBUTING](https://github.com/xbmc/repo-scripts/blob/master/CONTRIBUTING.md) document
- [x ] Each add-on submission should be a single commit with using the following style: [script.foo.bar] v1.0.0

### CODE REVIEW

File: script.artistslideshow/default.py [hunk 1]
```diff
@@ -1075,11 +997,302 @@ def _upgrade( self ):
...
+        inforoot = os.path.join( self.DATAROOT, 'ArtistInformation' )
+        imgroot = os.path.join( self.DATAROOT, 'ArtistSlideshow' )
+        infoarchiveroot = os.path.join( self.DATAROOT, 'archive', 'ArtistInformation' )
+        imgarchiveroot = os.path.join( self.DATAROOT, 'archive', 'ArtistSlideshow' )
+        artist_hashes = self._upgrade_get_artists_hashmap()
+        aDialog = xbmcgui.DialogProgressBG()
+        aDialog.create( smartUTF8(language(32013)), smartUTF8(language(32012)) )
+        try:
+            info_dirs, old_files = xbmcvfs.listdir( inforoot )
+        except Exception, e:
```
Comment (Rechi) [inline script.artistslideshow/default.py]: `except Exception as e:` is the preferred syntax as it is python3 compatible

File: script.artistslideshow/default.py [hunk 2]
```diff
@@ -1075,11 +997,302 @@ def _upgrade( self ):
...
+                
+
+    def _upgrade_artist_images( self ):
+        inforoot = os.path.join( self.DATAROOT, 'ArtistInformation' )
+        imgroot = os.path.join( self.DATAROOT, 'ArtistSlideshow' )
+        iDialog = xbmcgui.DialogProgressBG()
+        iDialog.create( smartUTF8(language(32013)), smartUTF8(language(32012)) )
+        try:
+            info_dirs, old_files = xbmcvfs.listdir( inforoot )
+        except Exception, e:
```
Comment (Rechi) [inline script.artistslideshow/default.py]: `except Exception as e:`

File: script.artistslideshow/resources/common/fileops.py [hunk 3]
```diff
@@ -95,6 +118,26 @@ def renameFile ( filename, newfilename ):
...
 
 
+def popenWithTimeout( command, timeout ):
+    log_lines = []
+    try:
+        p = subprocess.Popen( command, stdout=subprocess.PIPE, stderr=subprocess.PIPE )
+    except OSError:
+        log_lines.append( 'error finding external script, terminating' )
+        return False, log_lines
+    except Exception, e:
```
Comment (Rechi) [inline script.artistslideshow/resources/common/fileops.py]: `except Exception as e:`

File: script.artistslideshow/resources/common/fileops.py [hunk 4]
```diff
@@ -53,6 +57,25 @@ def deleteFile( filename ):
...
+def deleteFolder( foldername ):
+    log_lines = []
+    if _exists( foldername ):
+        try:
+            _rmdir( foldername )
+            log_lines.append( 'deleting folder %s' % foldername )
+        except IOError:
+            log_lines.append( 'unable to delete %s' % foldername )
+            return False, log_lines
+        except Exception, e:
```
Comment (Rechi) [inline script.artistslideshow/resources/common/fileops.py]: `except Exception as e:`

File: script.artistslideshow/default.py [hunk 5]
```diff
@@ -1075,11 +997,302 @@ def _upgrade( self ):
...
+            count = count + 1
+        iDialog.close()
+
+
+
+    def _upgrade_archive_folder( self, src, dst ):
+        lw.log( ['moving from %s to %s' % (src, dst)] )               
+        try:
+            folders, files = xbmcvfs.listdir( os.path.join( src, '' ) )
+        except Exception, e:
```
Comment (Rechi) [inline script.artistslideshow/default.py]: `except Exception as e:`

File: script.artistslideshow/default.py [hunk 6]
```diff
@@ -1075,11 +997,302 @@ def _upgrade( self ):
...
+    def _upgrade_get_artists_hashmap( self ):
+        hDialog = xbmcgui.DialogProgressBG()
+        hDialog.create( smartUTF8(language(32011)), smartUTF8(language(32012)) )
+        hashmap = _ordereddict()
+        response = xbmc.executeJSONRPC ( '{"jsonrpc":"2.0", "method":"AudioLibrary.GetArtists", "params":{"albumartistsonly":false, "sort":{"order":"ascending", "ignorearticle":true, "method":"artist"}},"id": 1}}' )
+        try:
+            artists_info = _json.loads(response)['result']['artists']
+        except (IndexError, KeyError, ValueError):
+            artists_info = []
+        except Exception, e:
```
Comment (Rechi) [inline script.artistslideshow/default.py]: `except Exception as e:`

File: script.artistslideshow/default.py [hunk 7]
```diff
@@ -1075,11 +997,302 @@ def _upgrade( self ):
...
+        return hashmap
+
+
+    def _upgrade_migratetolocal( self ):
+        mDialog = xbmcgui.DialogProgressBG()
+        mDialog.create( smartUTF8(language(32015)), smartUTF8(language(32012)) )
+        imgroot = os.path.join( self.DATAROOT, 'ArtistSlideshow' )
+        try:
+            img_dirs, old_files = xbmcvfs.listdir( imgroot )
+        except Exception, e:
```
Comment (Rechi) [inline script.artistslideshow/default.py]: `except Exception as e:`

File: script.artistslideshow/default.py [hunk 8]
```diff
@@ -1075,11 +997,302 @@ def _upgrade( self ):
...
+            if not exists:
+                imgarchive = os.path.join( self.DATAROOT, 'archive', 'ArtistSlideshow', img_dir_name )
+                self._upgrade_archive_folder( default_dir, imgarchive )
+                continue
+            local_dir = os.path.join( self.LOCALARTISTPATH, img_dir_name, self.FANARTFOLDER )
+            imgdb = os.path.join( local_dir, self.IMGDB )
+            exists, loglines = checkPath( os.path.join( local_dir, '' ) )
+            try:
+                throwaway, images = xbmcvfs.listdir( default_dir )
+            except Exception, e:
```
Comment (Rechi) [inline script.artistslideshow/default.py]: `except Exception as e:`

File: script.artistslideshow/default.py (hunk 1 above)
Comment (pkscout) [reply script.artistslideshow/default.py]: Thanks for the heads up.  I've been updating that as I update code that I actually run in Python3, but I hadn't checked to see if that syntax was backwards compatible.  Do I need to fix this for the current pull request, or can I just stick it in my todo list for the next update?

Comment (MartijnKaijser) [reply script.artistslideshow/default.py]: To do list is fine

**Comment by Lunatixz** [comment]
After @Rechi recommended changes it LGTM.

**Pull request merged**